components/SearchBox.vue

```diff
@@ -0,0 +1,179 @@
+        v-else-if="(filters.isLoading==true)"
+      >
+        <!-- Spinner-->
+        <div class="absolute right-1/2 bottom-1/2  transform translate-x-1/2 translate-y-1/2">
+          <div class="border-t-transparent border-solid animate-spin rounded-full border-blue-400 border-8 h-56 w-56" />
+        </div>
+      </tbody>
+
+      <tbody v-else>
+        <tr>
+          <td
+            colspan="13"
+            class="text-center py-4 border-b border-gray-200"
+          >
+            No Data Available
+          </td>
+        </tr>
+      </tbody>
+    </table>
+  </div>
+</template>
+
+<script setup>
+import { computed, ref, onMounted, watch } from 'vue'
+import { useSearchFilters } from '../store/searchfilters'
+
+const filters = useSearchFilters()
+
+// Reactive reference for rows
+const rows = ref([])
+const results = computed(() => filters.resultNum)
+
+// Selected from dropdown by user
+const selectedColumns = computed(() => filters.selectedColumns)
+// Columns to be displayed
+const filteredColumns = computed(() => {
+  return fixedColumns.filter((column) => selectedColumns.value.includes(column.name))
+})
+// Object to store the filter values entered by the user
+const columnFilters = computed(() => filters.filters)
+
+const fixedColumns = [
+  { name: 'Status', key: 'Status' },
+  { name: 'LastModifiedBy', key: 'Modified By' },
+  { name: 'NameRequestNumber', key: 'NR Number' },
+  { name: 'Names', key: 'Names' },
+  { name: 'ApplicantFirstName', key: 'Applicant First Name' },
+  { name: 'ApplicantLastName', key: 'Applicant Last Name' },
+  { name: 'NatureOfBusiness', key: 'Nature Of Business' },
+  { name: 'ConsentRequired', key: 'Consent Required' },
+  { name: 'Priority', key: 'Priority' },
+  { name: 'ClientNotification', key: 'Notified' },
+  { name: 'Submitted', key: 'Submitted' },
+  { name: 'LastUpdate', key: 'Last Update' },
+  { name: 'LastComment', key: 'Last Comment' }
+]
+
+const handleFilterChange = async () => {
+  // Update the Pinia store with the new filter values
+  await filters.updateFilters(columnFilters.value)
+
+  // Fetch rows based on the updated filters from the Pinia store
+  await filters.getRows() // Now, this uses the updated filters for the API call
+  rows.value = filters.rows
+  console.log(rows.value)
+}
+
+// When component is mounted, to display initial values from the table
+onMounted(async () => {
+  await filters.getRows()
+  rows.value = filters.rows
+})
+
+const dropdownOptions = {
+  Status: ['All', 'HOLD', 'INPROGRESS', 'DRAFT', 'EXPIRED',
+    'CANCELLED', 'APPROVED', 'CONDITIONAL', 'CONSUMED', 'REJECTED', 'COMPLETED'],
+  ConsentRequired: ['All', 'Yes', 'No', 'Received'],
+  Priority: ['All', 'Priority', 'Standard'],
+  ClientNotification: ['All', 'Notified', 'Not Notified'],
+  Submitted: ['All', 'Today', '7 days', '30 days', '90 days', '1 year', '3 years', '5 years', 'All', 'Custom'],
+  LastUpdate: ['All', 'Today', 'Yesterday', '2 days', '7 days', '30 days', 'All']
+}
```

Comment:
Are you going to need to relate these options to specific actions/filters? If so then I recommend using enums for the values, otherwise you're going to end up comparing strings.

Reply:
Using enums now

Reply:
Looks good.

(I did a double-take here, and researched it, but this appears to be the accepted way to get all values from enums.)